### PR TITLE
Initial pull request to update to new YT design

### PR DESCRIPTION
**Pull request by OiYouYeahYou**
Contains an initial refactoring effort and the first fix 

Re: #46 ( Dupes #48 #49 )

### CODE REVIEW

File: src/chrome/BackgroundScripts/s_CommonMethods.js [hunk 1]
```diff
@@ -1,175 +1,90 @@
...
+		// 		options.push( key );
+
+		// console.log( items );
 
-	if (root.children != null && returnValue == null) {
-		for (var i = 0; i < root.children.length; i++) {
-			var t = root.children[i].getAttribute(attributeName);
-			if (t != null) {
-				var c = 1;
 			}
```
Comment (Mattie432) [inline src/chrome/BackgroundScripts/s_CommonMethods.js]: Looks like extra bracket.

File: src/chrome/BackgroundScripts/s_CommonMethods.js [hunk 2]
```diff
@@ -1,175 +1,90 @@
...
- * 	set topLevel to true.
+ * Function to retrieve settings from googles online storage, then calls
+ * CallbackFunction with the result.
  */
-var returnValue;
-function searchAllChildrenFor(root, attributeName, attributeValue, topLevel) {
-	if (topLevel == true) {
-		returnValue = null;
-	}
+function getStoredChromeSettings( KeyArray, cb ) {
```
Comment (Mattie432) [inline src/chrome/BackgroundScripts/s_CommonMethods.js]: cb is never called.

**Comment by Mattie432** [comment]
Fantastic effort on the refactor, dead code and general cleanup! Much appreciated!

There does appear to be some errors. I've pointed out a few I noticed on quick glance, there may be a few more. As such i cannot verify the 'logo redirect fix'.

Can we split this out into two pull's one for the refactor, the second for YT update. Would make it easier to ensure nothing breaks.

**Comment by OiYouYeahYou** [comment]
I did try splitting out the PRs but couldn't figure out how to. As of commit 1e9e6d80a9d3f640d5d3989142edcd8d1a576d91 no bugs should be present.

Though I've done a quick push to fix what you have noticed

Addendum: Do you mind that my version iteration is `x.y.z` instead of `x.y`

**Comment by robin-gh2** [comment]
What is the status guys? A couple of days ago YT changed their design again en the 'Restore Old YT' doesn't work anymore.
An update of YouTweak would be much appreciated

**Comment by Deses** [comment]
Merge the branch already and push a goddamn update to the chrome store!

**Comment by Deses** [comment]
TBH, @OiYouYeahYou at this point I think you should just publish your fork to the Chrome Store and be done with it.

**Pull request closed without merging**